lib/resty/lmdb.lua

```diff
@@ -2,47 +2,126 @@ local _M = {}
 
 
 local transaction = require("resty.lmdb.transaction")
+local prefix = require("resty.lmdb.prefix")
 local status = require("resty.lmdb.status")
 
-do
-    local CACHED_TXN = transaction.begin(1)
 
+local assert = assert
+local prefix_page = prefix.page
+local get_phase = ngx.get_phase
+local ngx_sleep = ngx.sleep
 
-    function _M.get(key, db)
-        CACHED_TXN:reset()
-        CACHED_TXN:get(key, db)
-        local res, err = CACHED_TXN:commit()
-        if not res then
-            return nil, err
-        end
 
-        return CACHED_TXN[1].result
+local CACHED_TXN = transaction.begin(1)
+local CAN_YIELD_PHASES = {
```

Comment:
should we add `server_rewrite` `preread`?

Reply:
Does `server_rewrite` supports yielding?

Reply:
See: https://github.com/Kong/kong/blob/master/kong/tools/yield.lua#L14.
I think it is correct.

Reply:
Updated, thanks